client/opentracing/src/main/java/org/hawkular/apm/client/opentracing/TraceContext.java

```diff
@@ -104,7 +114,18 @@ public void endProcessingNode() {
             trace.setTransaction(getTransaction());
             trace.getNodes().add(node);
 
-            recorder.report(trace);
+            if (checkForSamplingProperties(node)) {
```

Comment:
I think the reporting level will need to be set when spans are created - and if the reporting level changes at a particular Span, it may require a sub-tracing context to be created to representing that sampling scope. That way, any inject performed on that or further child nodes, will inherit the reporting level from that sub-tree, whereas other nodes in the same trace fragment will use the reporting level of its parent - not sure if I have explained it well, so will try an example:

Root node in fragment is passed reporting level (or initialises it based on sampler if not based on extracted context). Say this reporting level is All.

If two child Spans (components) A and B are created, each inheriting trace context and reporting level. Then a further child Span is created below A and B, called A' and B'.

A' has its 'sampled' tag set to 0.

Both A' and B' are used to inject context in two separate external service invocations. The call from A' should pass reportingLevel None, while the call from B' will have reportingLevel All.

So the reportingLevel at each stage needs to be determined when the Span is created.